js/packages/screenshoter/src/get-target.js

```diff
@@ -37,27 +37,34 @@ async function getTarget({window, context, region, fully, scrollingMode, logger}
           await scrollingElement.scrollTo(utils.geometry.offsetNegative(elementLocation, scrollingLocation))
         }
       }
+      let returnRegion, scrollingElement, scroller
 
       if (fully) {
         const isScrollable = await element.isScrollable()
         // if element is scrollable, then take screenshot of the full element content, otherwise take screenshot of full element
-        const region = isScrollable ? null : await element.getRegion()
-        const scrollingElement = isScrollable ? element : await elementContext.getScrollingElement()
+        returnRegion = isScrollable ? null : await element.getRegion()
+        scrollingElement = isScrollable ? element : await elementContext.getScrollingElement()
         // css stitching could be applied only to root element of its context
         scrollingMode = scrollingMode === 'css' && !(await scrollingElement.isRoot()) ? 'mixed+' : scrollingMode
-        return {
-          context: elementContext,
-          region,
-          scroller: makeScroller({element: scrollingElement, scrollingMode, logger}),
-        }
       } else {
-        const scrollingElement = await context.getScrollingElement()
-        return {
-          context: elementContext,
-          region: await element.getRegion(),
-          scroller: makeScroller({element: scrollingElement, scrollingMode, logger}),
+        scrollingElement = await context.getScrollingElement()
+        returnRegion = await element.getRegion()
+      }
+      scroller = makeScroller({element: scrollingElement, scrollingMode, logger})
+      const isScrollerRoot = await scroller.element.isRoot()
+      if (!isScrollerRoot && returnRegion) {
+        const scrollerOffset = await scroller.getScrollOffset()
+        returnRegion = {
+          ...returnRegion,
+          x: returnRegion.x + scrollerOffset.x,
+          y: returnRegion.y + scrollerOffset.y,
         }
```

Comment:
We already have utility functions for this kind of geometry manipulation.
```suggestion
        returnRegion = utils.geometry.offset(returnRegion, scrollerOffset)
```